ui/src/main/java/com/theoplayer/android/ui/SeekBar.kt

```diff
@@ -27,32 +27,42 @@ fun SeekBar(
     val player = Player.current
     val currentTime = player?.currentTime?.toFloat() ?: 0.0f
     val seekable = player?.seekable ?: TimeRanges(listOf())
-    val valueRange = seekable.bounds ?: 0.0..0.0
 
+    val valueRange = remember(seekable) {
+        val bounds = seekable.bounds ?: 0.0..0.0
+        bounds.start.toFloat()..bounds.endInclusive.toFloat()
+    }
     var seekTime by remember { mutableStateOf<Float?>(null) }
     var wasPlayingBeforeSeek by remember { mutableStateOf(false) }
 
     Slider(
         modifier = modifier,
         colors = colors,
         value = seekTime ?: currentTime,
-        valueRange = valueRange.start.toFloat()..valueRange.endInclusive.toFloat(),
+        valueRange = valueRange,
         enabled = seekable.isNotEmpty(),
-        onValueChange = { time ->
-            seekTime = time
-            player?.player?.let {
-                if (!it.isPaused) {
-                    wasPlayingBeforeSeek = true
-                    it.pause()
+        onValueChange = remember {
+            { time ->
+                seekTime = time
+                player?.player?.let {
+                    if (!it.isPaused) {
+                        wasPlayingBeforeSeek = true
+                        it.pause()
+                    }
+                    it.currentTime = time.toDouble()
                 }
-                it.currentTime = time.toDouble()
             }
         },
-        onValueChangeFinished = {
-            seekTime = null
-            if (wasPlayingBeforeSeek) {
-                player?.player?.play()
-                wasPlayingBeforeSeek = false
+        // This needs to always be the *same* callback,
+        // otherwise Slider will reset its internal SliderState while dragging.
+        // https://github.com/androidx/androidx/blob/4d69c45e6361a2e5af77edc9f7f92af3d0db3877/compose/material3/material3/src/commonMain/kotlin/androidx/compose/material3/Slider.kt#L270-L282
+        onValueChangeFinished = remember {
```

Comment:
I'd say this is a regression in Compose Material 3. This used to work fine a year ago, and [all of the code samples](https://developer.android.com/reference/kotlin/androidx/compose/material3/package-summary#Slider(kotlin.Float,kotlin.Function1,androidx.compose.ui.Modifier,kotlin.Boolean,kotlin.Function0,androidx.compose.material3.SliderColors,androidx.compose.foundation.interaction.MutableInteractionSource,kotlin.Int,kotlin.Function1,kotlin.Function1,kotlin.ranges.ClosedFloatingPointRange)) simply pass `onValueChangeFinished` directly.

I'll see if I can file an issue upstream... on Monday. First up: weekend! 😁

Reply:
Ah, there's already [an open issue](https://issuetracker.google.com/u/1/issues/322269951) about it. And [one of the commenters found the same workaround](https://issuetracker.google.com/u/1/issues/322269951#comment8) by `remember`ing the callback. 🙂